packages/snap-tracker/src/Tracker.ts

```diff
@@ -211,6 +212,45 @@ export class Tracker {
 			return beaconEvent;
 		},
 
+		error: (data: TrackErrorEvent, siteId?: string): BeaconEvent | undefined => {
+			if (!data?.stack && !data?.message) {
+				// no console log
+				return;
+			}
+			let context = this.context;
+			if (siteId) {
+				context = deepmerge(context, {
+					context: {
+						website: {
+							trackingCode: siteId,
+						},
+					},
+				});
+			}
+			const payload = {
+				type: BeaconType.ERROR,
+				category: BeaconCategory.METRICS,
+				context,
+				event: {
+					type: data?.type || 'error',
+					userAgent: data?.userAgent || navigator.userAgent,
```

Comment:
We should not assume navigator or window are available here - we will be pulling these globals out in the near future.